src/main/java/org/apache/commons/io/TaggedIOException.java

```diff
@@ -89,6 +92,9 @@ public static boolean isTaggedWith(final Throwable throwable, final Object tag)
      * @param throwable an exception
      * @param tag tag object
      * @throws IOException original exception from the tagged decorator, if any
+     *
+     * checker issues false positive warning , this method doesn't throw
+     * null, getCause doesn't return null value as cause is-known/exists.
      */
     public static void throwCauseIfTaggedWith(final Throwable throwable, final Object tag)
             throws IOException {
```

Comment:
@mernst 
The circular documentation for `Throwable.getCause()` is little confusing. I referred this [post](https://stackoverflow.com/questions/6158817/exception-getcause-returning-null-when-trying-to-find-the-source-of-an-excepti), it was helpful to understand this method in case of chained exceptions. 
In `throwCauseIfTaggedWith` parameter `throwable` is non-null, does it mean that `cause` of exception exists and `getCause()` returns non-null value.

Reply:
I don't see any relationship between getCause() and the tag.

Reply:
Does return type of `getCause()` depend on `throwable` ? The working of `getCause()` is not much clear, so I am not able to decide if this is bug or false positive warning.

Reply:
`getCause` is declared at the end of the file; its compile-time type is `@Nullable IOException` and this does not depend on any compile-time information about `throwable`.

Reply:
Thanks Mike,
Non-null value for `cause` in `TaggedIOException` is set by invoking superclass i.e `IOExceptionWithCause` class constructor.
```
public TaggedIOException(final IOException original, final Serializable tag) {
        super(original.getMessage(), original);
        this.tag = tag;
}
```
`cause` of Exception is assigned value of `original` in super-class. Here `original` is non null, hence `cause` of `TaggedIOException` is non-null. This warning is false positive as `getCause()` returns non-null value. I have suppressed this warning in latest commit.